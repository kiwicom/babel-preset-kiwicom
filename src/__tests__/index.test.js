// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';
import * as babel from '@babel/core';

function transform(input) {
  const oldEnv = process.env.NODE_ENV;
  process.env.NODE_ENV = 'production';

  const transformedCode = babel.transform(input, {
    presets: [require('../index')],
  }).code;

  process.env.NODE_ENV = oldEnv;
  return transformedCode;
}

generateTestsFromFixtures(`${__dirname}/__fixtures__`, transform);
