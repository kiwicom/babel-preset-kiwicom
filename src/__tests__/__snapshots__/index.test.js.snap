// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: valid-code.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
// @flow

export default async function test(a: Object): Object {
  return {
    ...a,
    test_1: a?.b ?? 'default',
    test_2: !!__DEV__,
    test_3: await 'this',
  };
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = test;

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

async function test(a) {
  var _ref;

  return _objectSpread({}, a, {
    test_1: (_ref = a === null || a === void 0 ? void 0 : a.b) !== null && _ref !== void 0 ? _ref : 'default',
    test_2: !!(process.env.NODE_ENV !== "production"),
    test_3: await 'this'
  });
}
`;
